tests/test_sockets.py

```diff
@@ -692,6 +678,21 @@ async def handle(stream: SocketStream) -> None:
 
             tg.cancel_scope.cancel()
 
+    @skip_ipv6_mark
+    @pytest.mark.skipif(
+        sys.platform == "win32",
+        reason="Windows does not support interface name suffixes",
+    )
+    async def test_bind_link_local(self) -> None:
+        # Regression test for #554
+        for ifname, addresses in psutil.net_if_addrs().items():
+            for addr in addresses:
+                if addr.address.startswith("fe80::") and "%" in addr.address:
+                    async with await create_tcp_listener(local_host=addr.address):
+                        return
+
+        pytest.fail("Could not find a link-local IPv6 interface")
```

Comment:
a for loop with early return makes this test a bit odd, maybe a genexpr makes it a bit more obvious what's going on?

also you don't need .items() you can use .values()

```suggestion
        link_local_ipv6_address = next(
            (
                addr.address
                for addresses in psutil.net_if_addrs().values()
                for addr in addresses
                if addr.address.startswith("fe80::") and "%" in addr.address
            ),
            None,
        )

        if link_local_ipv6_address is None:
            pytest.fail("Could not find a link-local IPv6 interface")
            
        async with await create_tcp_listener(local_host=addr.address):
            pass
```